src/test/ui/suggestions/str-array-assignment.stderr

```diff
@@ -0,0 +1,44 @@
+error[E0308]: if and else have incompatible types
+  --> $DIR/str-array-assignment.rs:13:11
+   |
+13 |   let t = if true { s[..2] } else { s };
+   |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected str, found &str
+   |
+   = note: expected type `str`
+              found type `&str`
+
+error[E0308]: mismatched types
+  --> $DIR/str-array-assignment.rs:17:27
+   |
+11 | fn main() { //~ NOTE expected `()` because of default return type
+   |           - expected `()` because of default return type
```

Comment:
this looks dubious. Could you look into this?

Reply:
Yeah, I noticed that too. Still digging.

Reply:
it's also already the case on nightly. Feel free to ignore this and open an issue.

Reply:
#46302